README.md
# Arm_level_Alteration
Calculate percentage chromosome copy number alteration (Gain, Loss, LOH) based on Affymetrix Oncoscan Assay Percentage alteration defined based on the chromosome length covered by oncoscan


history.txt
    1  sudo mount -t drvfs '\\huge.ad.hcuge.ch\ROOT\HOME\P'
    2  sudo mount -t drvfs '\\huge.ad.hcuge.ch\ROOT\HOME\P' /mnt/v/
    3  sudo mount -t drvfs '\\huge.ad.hcuge.ch\ROOT\HOME\P' /mnt/v
    4  cd /mnt/c/Users/prac/
    5  ls
    6  cd Desktop/
    7  ls
    8  cd ..
    9  ls
   10  cd Downloads/
   11  ls
   12  cd ../Documents/
   13  ls
   14  cd R/
   15  ls
   16  cd ../
   17  ls
   18  mkdir scratch
   19  cd scratch/
   20  cp ~/Psize.txt .
   21  cp ~/Qsite.txt .
   22  cp ~/chromosome_stat_formatted.txt .
   23  l
   24  cd ~/
   25  ls
   26  mount //huge/ROOT/HOME/P/PRAC /mnt/v/
   27  sudo mount //huge/ROOT/HOME/P/PRAC /mnt/v/
   28  sudo apt-get install cifs-utils
   29  sudo mount -t  //huge/ROOT/HOME/P/PRAC /mnt/v/
   30  sudo mount -t  //huge/ROOT/HOME/P/PRAC /mnt/v
   31  man mount
   32  sudo mount -t cifs //huge/ROOT/HOME/P/PRAC /mnt/v
   33  man mount
   34  sudo mount -t cifs -o vers=2.1,nodfs //huge/ROOT/HOME/P/PRAC /mnt/v
   35  sudo mount -t cifs -o vers=2.1,nodfs,username=prac,password=Rkw9rchi,file_mode=0777,dir_mode=0777 //huge/ROOT/HOME/P/PRAC /mnt/v
   36  cat /proc/filesystems 
   37  sudo modprobe cifs
   38  exit
   39  history 
   40  cat /proc/filesystems 
   41  sudo modprobe cifs
   42  sudo mount -t drvfs //huge/ROOT/HOME/P/PRAC /mnt/v
   43  sudo apt-get install samba
   44  sudo mount -t cifs -o vers=2.1,nodfs,username=prac,password=Rkw9rchi,file_mode=0777,dir_mode=0777 //huge/ROOT/HOME/P/PRAC /mnt/v
   45  sudo mount -o vers=2.1,nodfs,username=prac,password=Rkw9rchi,file_mode=0777,dir_mode=0777 //huge/ROOT/HOME/P/PRAC /mnt/v
   46  sudo modprobe cifs
   47  vim 
   48  vim
   49  ls
   50  paste -d "\t" chr_name.txt chr_arm.txt chr_length.txt 
   51  paste -d "\t" chr_name.txt chr_arm.txt chr_length.txt >chromosome_arm_lengths_oncoscan_covered.txt
   52  vim chromosome_arm_lengths_oncoscan_covered.txt 
   53  ls
   54  rm chr_*
   55  ls
   56  rm chromosome_stat*
   57  ls
   58  rm Psize.txt Qsite.txt 
   59  ls
   60  awk '{print $1}' chromosome_arm_lengths_oncoscan_covered.txt 
   61  vim
   62  awk '{print $2}' chromosome_arm_lengths_oncoscan_covered.txt 
   63  vim
   64  awk '{print $3}' chromosome_arm_lengths_oncoscan_covered.txt 
   65  vim
   66  pwd
   67  ls
   68  mv chromosome_arm_lengths_oncoscan_covered.txt /mnt/c/Users/prac/Documents/scratch/
   69  ls
   70  exit
   71  ls
   72  cd /mnt/c/Users/prac/Desktop/
   73  ls
   74  head scna_oncoscan_summary_16jan_2019.dat 
   75  vim scna_oncoscan_summary_16jan_2019.dat 
   76  head scna_oncoscan_summary_16jan_2019.dat 
   77  awk '{print $1}' scna_oncoscan_summary_16jan_2019.dat 
   78  awk '{print $1}' scna_oncoscan_summary_16jan_2019.dat | sort -nk1 | uniq
   79  vim 
   80  ls
   81  cat OncoScan.na33.r1.chromStats.tsv 
   82  cat scna_oncoscan_summary_16jan_2019.dat 
   83  grep -P "^22" scna_oncoscan_summary_16jan_2019.dat 
   84  grep -P "^22\tp" scna_oncoscan_summary_16jan_2019.dat 
   85  grep -P "^20\tp" scna_oncoscan_summary_16jan_2019.dat 
   86  grep -P "^20\tp" scna_oncoscan_summary_16jan_2019.dat  | sort -nk4
   87  ls
   88  cd genelists-wLocation/
   89  ls
   90  cd ../
   91  ls
   92  cd genelists-wLocation/
   93  ls
   94  ls sc
   95  ls
   96  cd ../
   97  cat scna_oncoscan_summary_16jan_2019.dat 
   98  cd genelists-wLocation/
   99  ls
  100  grep Gain H18013536_A_gene_list_full_location.txt | awk -F "\t" '{print $NF}'
  101  grep -P "Gain\t20\t" H18013536_A_gene_list_full_location.txt | awk -F "\t" '{print $NF}'
  102  grep -P "Gain\t20\t" H18013536_A_gene_list_full_location.txt | awk -F "\t" '{print $5,$6, $NF}'
  103  head -2 ../scna_oncoscan_summary_16jan_2019.dat 
  104  grep -P "^20" ../scna_oncoscan_summary_16jan_2019.dat | grep H18013536
  105  grep -P "Gain\t20\t" H18013536_A_gene_list_full_location.txt | awk -F "\t" '{print $5,$6,$7, $NF}'
  106  bc -l
  107  kill %1
  108  head -2 H18013536_A_gene_list_full_location.txt | awk '{print $4,$5,$6,$7,$NF}'
  109  head -2 H18013536_A_gene_list_full_location.txt | awk '{print $3,$4,$5,$6,$7,$NF}'
  110  head -2 H18013536_A_gene_list_full_location.txt | awk -F "\t" '{print $3,$4,$5,$6,$7,$NF}'
  111  grep -P "Gain\t20\t" H18013536_A_gene_list_full_location.txt | awk -F "\t" '{print $5,$6, $NF}'
  112  exit
  113  ls
  114  cd /mnt/c/Users/prac/Desktop/
  115  ls
  116  cd genelists-wLocation/
  117  ls
  118  ls -lrt
  119  vim chr_alt.sh 
  120  ls
  121  ls -lrt
  122  vim chr_alt_dis.R 
  123  vim chr_alt.sh 
  124  cp chr_alt.sh chr_alt_23jan.sh
  125  vim chr_alt_23jan.sh 
  126  ./chr_alt_23jan.sh 
  127  ls -lrtr
  128  ./chr_alt_23jan.sh 
  129  vim chr_alt_23jan.sh 
  130  ls
  131  vim chr_alt_23jan.sh 
  132  ./chr_alt_23jan.sh 
  133  vim chr_alt_23jan.sh 
  134  head -2 H18013536_A_gene_list_full_location.txt
  135  vim chr_alt_23jan.sh 
  136  ./chr_alt_23jan.sh 
  137  vim chr_alt_23jan.sh 
  138  ./chr_alt_23jan.sh 
  139  kill %1
  140  for i in `grep -v "File" H18013536_A_gene_list_full_location.txt`; do print $i ; done
  141  grep -v "File" H18013536_A_gene_list_full_location.txt
  142  grep -v "File" H18013536_A_gene_list_full_location.txt | head -1
  143  grep -v "File" H18013536_A_gene_list_full_location.txt | head -1 | awk -F "\t" '{print $NF}'
  144  grep -v "File" H18013536_A_gene_list_full_location.txt | awk -F "\t" '{print $NF}'
  145  ./chr_alt_23jan.sh 
  146  for l in `grep -v "File" H18013536_A_gene_list_full_location.txt`; do echo $l ; done
  147  for l in `cat H18013536_A_gene_list_full_location.txt`; do echo $l ; done
  148  cat H18013536_A_gene_list_full_location.txt
  149  cat H18013536_A_gene_list_full_location.txt | head -1
  150  for i in `cat H18013536_A_gene_list_full_location.txt`; do echo $i ; done
  151  for i in $(cat H18013536_A_gene_list_full_location.txt); do echo $i ; done
  152  for i in $(head -1 H18013536_A_gene_list_full_location.txt); do echo $i ; done
  153  for i in `head -1 H18013536_A_gene_list_full_location.txt`; do echo $i ; done

plot_dist.R
# test code to validate predictions from chr_arm_level_alt.R
source("arm_level_alt.R")

cutoff <- read.table("cutoffs.dat", 
                             sep = " ", header = TRUE)

for (i in 1:nrow(cutoff))
{
segment_thr<-as.numeric(cutoff$filt[i])
min_gap<-as.numeric(cutoff$smooth[i])

# List all files in the oncoscan_files directory

file_list<-list.files(path = "onscan_files/", pattern = NULL, all.files = FALSE,
                      full.names = TRUE, recursive = FALSE,
                      ignore.case = FALSE, include.dirs = FALSE, no.. = FALSE)


#get the chr_alt_summary for all samples based on the specified cut offs
lapply(file_list, chr_alt_summary, segment_thr=segment_thr,min_gap=min_gap)
}
#####################
# Plot distribution

library(ggplot2)
library(fitdistrplus)

#################################
# List all files in the chr_alt_out directory

chr_alt_out<-list.files(path = "./results/", pattern = NULL, all.files = FALSE,
                      full.names = TRUE, recursive = FALSE,
                      ignore.case = FALSE, include.dirs = FALSE, no.. = FALSE)
#################################
# readFile
#################################

readFile <- function(filename) {
  df <- read.csv(filename, header = FALSE, sep = " ")
  df <-df[,c(2,3,4,5,6,1)]
  
  colnames(df)<-c("CHR","GAIN","AMPLI","LOSS","LOH", "Filename")
  
  df[df == 0] <- NA
  
#  df$CHR<-gsub("p|q", "", df$CHR)
  
return(df)
}
################################
#lONG

df1<-readFile(chr_alt_out[1])
df2<-readFile(chr_alt_out[2])
df3<-readFile(chr_alt_out[3])
df4<-readFile(chr_alt_out[4])
df5<-readFile(chr_alt_out[5])
df6<-readFile(chr_alt_out[6])
df7<-readFile(chr_alt_out[7])
df8<-readFile(chr_alt_out[8])
df9<-readFile(chr_alt_out[9])
df10<-readFile(chr_alt_out[10])


ggplot() + 
  geom_density(data=df1, aes(x=GAIN, col='10k_1MB')) + 
  geom_density(data=df2, aes(x=GAIN, col='10k_10Mb')) +
  geom_density(data=df3, aes(x=GAIN, col='10k_1K')) +
  geom_density(data=df4, aes(x=GAIN, col='10k_300k')) +
  geom_density(data=df5, aes(x=GAIN, col='300k_1Mb')) +
  geom_density(data=df6, aes(x=GAIN, col='300k_10Mb'))+
  geom_density(data=df7, aes(x=GAIN, col='300k_300K')) +
  geom_density(data=df8, aes(x=GAIN, col='3Mb_1Mb')) +
  geom_density(data=df9, aes(x=GAIN, col='3Mb_10Mb')) +
  geom_density(data=df10, aes(x=GAIN, col='3Mb_300K'))
  
  
  

ggplot() + 
  geom_density(data=df1, aes(x=LOSS, col='10k_1MB')) + 
  geom_density(data=df2, aes(x=LOSS, col='10k_10Mb')) +
  geom_density(data=df3, aes(x=LOSS, col='10k_1K')) +
  geom_density(data=df4, aes(x=LOSS, col='10k_300k')) +
  geom_density(data=df5, aes(x=LOSS, col='300k_1Mb')) +
  geom_density(data=df6, aes(x=LOSS, col='300k_10Mb'))+
  geom_density(data=df7, aes(x=LOSS, col='300k_300K')) +
  geom_density(data=df8, aes(x=LOSS, col='3Mb_1Mb')) +
  geom_density(data=df9, aes(x=LOSS, col='3Mb_10Mb')) +
  geom_density(data=df10, aes(x=LOSS, col='3Mb_300K'))


################################
#SEG
#lONG

df1<-readFile(chr_alt_out[11])
df2<-readFile(chr_alt_out[12])
df3<-readFile(chr_alt_out[13])
df4<-readFile(chr_alt_out[14])
df5<-readFile(chr_alt_out[15])
df6<-readFile(chr_alt_out[16])
df7<-readFile(chr_alt_out[17])
df8<-readFile(chr_alt_out[18])
df9<-readFile(chr_alt_out[19])
df10<-readFile(chr_alt_out[20])


ggplot() + 
  geom_density(data=df1, aes(x=GAIN, col='10k_1MB')) + 
  geom_density(data=df2, aes(x=GAIN, col='10k_10Mb')) +
  geom_density(data=df3, aes(x=GAIN, col='10k_1K')) +
  geom_density(data=df4, aes(x=GAIN, col='10k_300k')) +
  geom_density(data=df5, aes(x=GAIN, col='300k_1Mb')) +
  geom_density(data=df6, aes(x=GAIN, col='300k_10Mb'))+
  geom_density(data=df7, aes(x=GAIN, col='300k_300K')) +
  geom_density(data=df8, aes(x=GAIN, col='3Mb_1Mb')) +
  geom_density(data=df9, aes(x=GAIN, col='3Mb_10Mb')) +
  geom_density(data=df10, aes(x=GAIN, col='3Mb_300K'))




ggplot() + 
  geom_density(data=df1, aes(x=LOSS, col='10k_1MB')) + 
  geom_density(data=df2, aes(x=LOSS, col='10k_10Mb')) +
  geom_density(data=df3, aes(x=LOSS, col='10k_1K')) +
  geom_density(data=df4, aes(x=LOSS, col='10k_300k')) +
  geom_density(data=df5, aes(x=LOSS, col='300k_1Mb')) +
  geom_density(data=df6, aes(x=LOSS, col='300k_10Mb'))+
  geom_density(data=df7, aes(x=LOSS, col='300k_300K')) +
  geom_density(data=df8, aes(x=LOSS, col='3Mb_1Mb')) +
  geom_density(data=df9, aes(x=LOSS, col='3Mb_10Mb')) +
  geom_density(data=df10, aes(x=LOSS, col='3Mb_300K'))

##########################
#SUM
#lONG

df1<-readFile(chr_alt_out[21])
df2<-readFile(chr_alt_out[22])
df3<-readFile(chr_alt_out[23])
df4<-readFile(chr_alt_out[24])
df5<-readFile(chr_alt_out[25])
df6<-readFile(chr_alt_out[26])
df7<-readFile(chr_alt_out[27])
df8<-readFile(chr_alt_out[28])
df9<-readFile(chr_alt_out[29])
df10<-readFile(chr_alt_out[30])


ggplot() + 
  geom_density(data=df1, aes(x=GAIN, col='10k_1MB')) + 
  geom_density(data=df2, aes(x=GAIN, col='10k_10Mb')) +
  geom_density(data=df3, aes(x=GAIN, col='10k_1K')) +
  geom_density(data=df4, aes(x=GAIN, col='10k_300k')) +
  geom_density(data=df5, aes(x=GAIN, col='300k_1Mb')) +
  geom_density(data=df6, aes(x=GAIN, col='300k_10Mb'))+
  geom_density(data=df7, aes(x=GAIN, col='300k_300K')) +
  geom_density(data=df8, aes(x=GAIN, col='3Mb_1Mb')) +
  geom_density(data=df9, aes(x=GAIN, col='3Mb_10Mb')) +
  geom_density(data=df10, aes(x=GAIN, col='3Mb_300K'))




ggplot() + 
  geom_density(data=df1, aes(x=LOSS, col='10k_1MB')) + 
  geom_density(data=df2, aes(x=LOSS, col='10k_10Mb')) +
  geom_density(data=df3, aes(x=LOSS, col='10k_1K')) +
  geom_density(data=df4, aes(x=LOSS, col='10k_300k')) +
  geom_density(data=df5, aes(x=LOSS, col='300k_1Mb')) +
  geom_density(data=df6, aes(x=LOSS, col='300k_10Mb'))+
  geom_density(data=df7, aes(x=LOSS, col='300k_300K')) +
  geom_density(data=df8, aes(x=LOSS, col='3Mb_1Mb')) +
  geom_density(data=df9, aes(x=LOSS, col='3Mb_10Mb')) +
  geom_density(data=df10, aes(x=LOSS, col='3Mb_300K'))




#f <- fitdist(data$GAIN/100, 'beta', method='mme', start=list(shape1=0.001, shape2=0.001))
#f$estimate
#ratio=log(f$estimate[1]/f$estimate[2])
#ratio

############################################################
#################################
# List all files in the chr_alt_out directory

results_10k_300k<-list.files(path = "./tmp", pattern = NULL, all.files = FALSE,
                        full.names = TRUE, recursive = FALSE,
                        ignore.case = FALSE, include.dirs = FALSE, no.. = FALSE)

long<-readFile(chr_alt_out[4])
sum<-readFile(chr_alt_out[24])
seg<-readFile(chr_alt_out[14])

chr_table$length<-chr_table$Arm_end-chr_table$Arm_str


# Plot number of segmets vs %alteration based on sum
boxplot(split(sum$GAIN,seg$GAIN), main="Gain: trim 10k, smooth 300k ",
        xlab="Number of segments gained", ylab="Percentage Gain")

plot(sum$GAIN,seg$GAIN,xlab="Percentage Gain", ylab="Number of segments gained")

boxplot(split(sum$LOSS,seg$LOSS), main="LOSS: trim 10k, smooth 300k ",
        xlab="Number of segments lost", ylab="Percentage Alteration")

plot(sum$LOSS,seg$LOSS,xlab="Percentage LOSS", ylab="Number of segments Lost")


# Normalize the seg lengths by arm length
for(i in 1:nrow(chr_table))
{
  for(j in 1:nrow(seg))
  {
    if (chr_table$Names[i]==seg$CHR[j])
    {
      seg$GAIN_norm[j]=seg$GAIN[j]/chr_table$length[i]
      seg$LOSS_norm[j]=seg$LOSS[j]/chr_table$length[i]
    }
  }
}

boxplot(split(sum$GAIN,seg$GAIN_norm), main="Gain: trim 10k, smooth 300k ",
        xlab="Number of segments", ylab="Percentage Alteration")

boxplot(split(sum$LOSS,seg$LOSS_norm), main="LOSS: trim 10k, smooth 300k ",
        xlab="Number of segments", ylab="Percentage Alteration")

plot(sum$GAIN,seg$GAIN_norm,xlab="Percentage Alteration", ylab="Number of segments")

plot(sum$LOSS,seg$LOSS_norm,xlab="Percentage Alteration", ylab="Number of segments")

#######################3
# Correlation sum vs long
x_g<-sum$GAIN
y_g<-long$GAIN

cor(x_g, y_g, use = "complete.obs")

plot(sum$GAIN,long$GAIN,xlab="Percentage GAIN (SUM)", ylab="Percentage GAIN (Longest)")

x_l<-sum$LOSS
y_l<-long$LOSS

cor(x_l, y_l, use = "complete.obs")

plot(sum$LOSS,long$LOSS,xlab="Percentage LOSS (SUM)", ylab="Percentage LOSS (Longest)")
